packages/asset/contracts/AssetReveal.sol

```diff
@@ -0,0 +1,266 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.8.18;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/cryptography/EIP712Upgradeable.sol";
+import "./libraries/TokenIdUtils.sol";
+import "./AuthValidator.sol";
+import "./ERC2771Handler.sol";
+import "./interfaces/IAsset.sol";
+import "./interfaces/IAssetReveal.sol";
+
+/// @title AssetReveal
+/// @author The Sandbox
+/// @notice Contract for burning and revealing assets
+contract AssetReveal is
+    IAssetReveal,
+    Initializable,
+    ERC2771Handler,
+    EIP712Upgradeable
+{
+    using TokenIdUtils for uint256;
+    IAsset private assetContract;
+    AuthValidator private authValidator;
+
+    // mapping of creator to asset id to asset's reveal nonce
+    mapping(address => mapping(uint256 => uint16)) revealNonces;
+
+    string public constant name = "Sandbox Asset Reveal";
+    string public constant version = "1.0";
```

Comment:
idem

Reply:
Lets discuss that in your previous comment. Personally I don't see a reason why this shouldn't be a constant declared like that, it makes it easy to manage, everytime we need to re-deploy we can change the version here rather than referring to some external file to check what to use and what versions were deployed in the past.

Reply:
you don't necessarely need to change the version when updating the implementation (this version is for the sig and it's there even for the non upgradedeable 712)